react-native-reanimated.d.ts

```diff
@@ -244,10 +244,28 @@ declare module 'react-native-reanimated' {
 
     export function createAnimatedComponent<P extends object>(
       component: ComponentClass<P>
-    ): ComponentClass<AnimateProps<P>>;
+    ): ComponentClass<
+      AnimateProps<
+        P &
+          ('style' extends keyof P
+            ? {}
+            : {
+                style?: StyleProp<ViewStyle>
```

Comment:
I don't actually know what Reanimated is doing under the hood but per #1717 it works to add an animated `style` prop to eg an `<AnimatedCircle>` (where `<Circle>` does not have a `style` prop). So I guessed here that the `style` prop exposed by Reanimated is compatible with `ViewStyle`